components/eamxx/src/share/io/scorpio_output.cpp

```diff
@@ -839,10 +839,28 @@ void AtmosphereOutput::register_views()
 
     // Now create and store a dev view to track the averaging count for this layout (if we are tracking)
     // We don't need to track average counts for files that are not tracking the time dim
+    set_avg_cnt_tracking(name,"",layout);
+  }
+
+  // Initialize the local views
+  reset_dev_views();
+}
+/* ---------------------------------------------------------- */
+void AtmosphereOutput::set_avg_cnt_tracking(const std::string& name, const std::string& name_ext, const FieldLayout& layout)
+{
+    // Make sure this field "name" hasn't already been regsitered with avg_cnt tracking.
+    // Note, we check this because some diagnostics need to have their own tracking which
+    // is created at the 'create_diagnostics' function.
+    if (m_field_to_avg_cnt_map.count(name)>0) {
+      return;
+    }
+    // Now create and store a dev view to track the averaging count for this layout (if we are tracking)
+    // We don't need to track average counts for files that are not tracking the time dim
+    const auto size = layout.size();
     const auto tags = layout.tags();
     auto lt = get_layout_type(tags);
     if (m_add_time_dim && m_track_avg_cnt) {
-      std::string avg_cnt_name = "avg_count_" + e2str(lt);
+      std::string avg_cnt_name = "avg_count_" + e2str(lt) + "_" + name_ext;
```

Comment:
If `name_ext=""`, which is the case for all but the `FieldAtPressureLevel` diagnostic, you get two underscores in the name (one here, and one from the for loop below). Can we jsut do `+ e2str(lt) + name_ext`, and put the underscore in the `name_ext` when it's computed (if not empty)?

Reply:
Also, maybe `name_ext` is not super self-explanatory. How about "avg_cnt_suffix"?

Reply:
Also, since we add the tag names to `avg_cnt_name` below, we can keep the name shorter and not add `e2str(lt)` maybe? It doesn't add any info not already in the list of tags...

Reply:
all good points.  I'll make the changes..